src/Core/Managed/Shared/Channel/TelemetryBuffer.cs

```diff
@@ -46,7 +48,30 @@ public int Capacity
                 this.capacity = value;
             }
         }
-        
+
+        /// <summary>
+        /// Gets or sets the maximum number of telemetry items that can be in the backlog to send. Items will be dropped
+        /// once this limit is hit.
+        /// </summary>        
+        public int MaximumUnsentBacklogSize
+        {
+            get
+            {
+                return this.maximumUnsentBacklogSize;
+            }
+
+            set
+            {
+                if (value < 1)
```

Comment:
do you have a number in mind? Does one hundred thousand sounds okay? Should it be different for inmemory vs serverchannels?